service/history/shard/context_impl.go

```diff
@@ -383,16 +383,23 @@ func (s *ContextImpl) UpdateReplicationQueueReaderState(
 // UpdateRemoteClusterInfo deprecated
 // Deprecated use UpdateRemoteReaderInfo in the future instead
 func (s *ContextImpl) UpdateRemoteClusterInfo(
-	cluster string,
+	clusterName string,
 	ackTaskID int64,
 	ackTimestamp time.Time,
 ) {
 	s.wLock()
 	defer s.wUnlock()
 
-	remoteClusterInfo := s.getOrUpdateRemoteClusterInfoLocked(cluster)
-	remoteClusterInfo.AckedReplicationTaskIDs[s.shardID] = ackTaskID
-	remoteClusterInfo.AckedReplicationTimestamps[s.shardID] = ackTimestamp
+	clusterInfo := s.clusterMetadata.GetAllClusterInfo()
+	remoteClusterInfo := s.getOrUpdateRemoteClusterInfoLocked(clusterName)
+	for _, remoteShardID := range common.MapShardID(
+		clusterInfo[s.clusterMetadata.GetCurrentClusterName()].ShardCount,
+		clusterInfo[clusterName].ShardCount,
+		s.shardID,
+	) {
+		remoteClusterInfo.AckedReplicationTaskIDs[remoteShardID] = ackTaskID
+		remoteClusterInfo.AckedReplicationTimestamps[remoteShardID] = ackTimestamp
```

Comment:
One thing I am not following, why the acked task id can be used across different share ids?

Reply:
for polling mode:
targer cluster's 1 poller may poll & ack on behalf of >= 1 shards, here source cluster should update all relevant shard's ack task IDs

for streaming mode:
each target shard will has its own grpc stream & update ack using reader ID (cluster ID + shard ID)